sender.go

```diff
@@ -26,6 +31,18 @@ type Sender struct {
 // additional messages can be sent while the current goroutine is waiting
 // for the confirmation.
 func (s *Sender) Send(ctx context.Context, msg *Message) error {
+	select {
+	case <-s.link.detached:
```

Comment:
Do we need a similar check on the `Receiver` to return `ErrLinkClosed`?

Reply:
I just updated in the other comment - I think we should do a similar thing here but perhaps have a new error code (ErrLinkDetached) so we can distinguish between explicit close by caller, and close from service (ie: detach)

Reply:
Receiver seems slightly more complicated - I think we should design this a bit better with Receiver in mind in a future PR.